File: cli/log.go
package main

import (
	"fmt"

	"github.com/fatih/color"
)

func Log(message string) {
	fmt.Println(message)
}

func LogInfo(message string) {
	color.Blue("i %s", message)
}

func LogDebug(message string) {
	color.Cyan("> %s", message)
}

func LogHighlight(message string) {
	color.Magenta("%s", message)
}

func LogSuccess(message string) {
	color.Green("✔ %s", message)
}

func LogError(message string) {
	color.Red("✗ %s", message)
}

func LogWarn(message string) {
	color.Yellow("! %s", message)
}

File: app/backend/log.ts
import chalk from "chalk";

export function log(logContent: string): void {
  // If a third-party logging service was used,
  // this would be point to hook in that integration.
  console.log(logContent);
}

// Show a styled highlight log message
export function logHighlight(message: string): void {
  console.log(chalk.hex("#D2A6FF")(message));
}

// Show a styled info log message
export function logInfo(message: string): void {
  log(chalk.blue(`i ${message}`));
}

// Show a styled success log message
export function logSuccess(message: string): void {
  log(chalk.green(`✔ ${message}`));
}

// Show a styled warning log message
export function logWarn(message: string): void {
  log(chalk.hex("#FFA22E")(`! ${message}`));
}

// Show a styled error log message
export function logError(message: string): void {
  log(chalk.red(`✗ ${message}`));
}

// Show a styled debug log message
export function logDebug(message: string): void {
  log(chalk.hex("#23C5B0")(`> ${message}`));
}
